python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -568,7 +568,7 @@ def reserve_and_allocate_spread(
         for bundles in strict_spreads:
             # Try to pack as many bundles of this group as possible on existing
             # nodes. The remaining will be allocated on new nodes.
-            unfulfilled, node_resources = get_bin_pack_residual(
+            unfulfilled, updated_node_resources = get_bin_pack_residual(
                 node_resources, bundles, strict_spread=True
             )
```

Comment:
If a strict spread bundle cannot be scheduled currently, we should ignore it. Therefore, we can't override the `node_resources` here. Use a new `updated_node_resources` instead.